apps/api-extractor/src/extractor/Extractor.ts

```diff
@@ -442,4 +457,47 @@ export class Extractor {
     }
     return path.relative(this._absoluteRootFolder, absolutePath).replace(/\\/g, '/');
   }
+
+  private _updateCommandLineForTypescriptLibPackage(
+    commandLine: ts.ParsedCommandLine,
+    options: IExtractorOptions
+  ): void {
+    if (options.typescriptLibPackagePath) {
+      commandLine.options.noLib = true;
+      const compilerLibDirectory: string = path.join(options.typescriptLibPackagePath, 'lib');
+
+      let foundBaseLib: boolean = false;
+      const filesToAdd: string[]  = [];
+      for (const libFilename of commandLine.options.lib || []) {
+        if (libFilename === 'lib.d.ts') {
+          // Ignore the default lib - it'll get added later
+          continue;
+        }
+
+        if (libFilename === 'lib.es5.d.ts' || libFilename === 'lib.es6.d.ts') {
```

Comment:
> 'lib.es5.d.ts' [](start = 27, length = 15)

Should these filenames go in a constant?  Should the comparison be case insensitive?

Reply:
It might be helpful to call 'lib.d.ts' the DEFAULT_BUILTIN_LIBRARY and the others OTHER_BUILTIN_LIBRARIES[].  This way it will be easier for people to guess how to add `lib.es7.d.ts` when that fails.

---
In reply to: [219345885](https://github.com/Microsoft/web-build-tools/pull/832#discussion_r219345885) [](ancestors = 219345885)